### PR TITLE
Enumerable Methods

### PR DESCRIPTION
**Pull request by pountzas**
the complete Enumerable Methods with all the requirements
#my_each
#my_each_with_index.
#my_select.
#my_all?
#my_any?
#my_none?
#my_count
#my_map
#my_inject
#my_inject
#my_inject, e.g. multiply_els()
#my_map method to take a proc instead.
#my_map method to take either a proc or a block.

### CODE REVIEW

File: enumerables.rb [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    arr.my_inject { |i, j| i * j }
+  end
+end
+
+array = %w[asd asdd qweee 2 3 4 5]
+num = [2, 4, 5]
+
+array.my_each(array) { |item| puts item }
+array.my_each_with_index(array) { |item, _index| puts "asdasd #{item}" }
+# num.my_select(array) { |item| puts item  }
```
Comment (cyonii) [inline enumerables.rb]: Please remove commented lines of code. [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

**Review by cyonii — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️✍️

Hello team, you have shown great effort in this project.
## To Highlight
- Good job having your linters configured and all green 🟢
- Great job keeping your project clean 

## To Improve
- [x] The issue link in your README is broken.

You have great effort so far, however, there are major issues you have to fix in order for your project to get approved. Below are the failures reported from the tests run against your code.

> Kindly take your time and look into this [documentary](https://ruby-doc.org/core-2.6.5/Enumerable.html), you can use the test cases used in this documentation since the same behaviours are expected of your code. 

**Failed examples:**

- [ ] `#my_each` returns an Enumerator if no block is given
- [ ] `#my_each` does not mutate the original array
- [ ] `#my_each` when a block is given, and when self is an array, it returns the array itself after calling the given block once for each element in self
- [ ] `#my_each` when a block is given, and when self is an array, it behaves like `#each` and call the given block once for each element in self
- [ ] `#my_each` when a block is given, and when self is a range, it returns the range itself after calling the given block once for each element in self
- [ ] `#my_each` when a block is given, and when self is a range, it behaves like `#each` and call the given block once for each element in self
- [ ] `#my_each` when a block is given, and when self is a hash, it behaves like `#each` and call the given block once for each element in self
- [ ] `#my_each_with_index` returns an enumerator if no block is given
- [ ] `#my_each_with_index` does not mutate the original array
- [ ] `#my_each_with_index` when a block is given, and when self is an array, it returns the array itself after calling the given block once for each element in self
- [ ] `#my_each_with_index` when a block is given, and when self is an array, it behaves like `#each_with_index` and call the given block once for each element in self
- [ ] `#my_each_with_index` when a block is given, and when self is a range, it returns the range itself after calling the given block once for each element in self
- [ ] `#my_each_with_index` when a block is given, and when self is a range, it behaves like `#each_with_index` and call the given block once for each element in self
- [ ] `#my_each_with_index` when a block is given, and when self is a hash, it behaves like `#each_with_index` and call the given block once for each element in self
- [ ] `#my_select` returns an array containing all elements of enum for which the given block returns a true value
- [ ] `#my_select` returns an array containing all elements of enum for which the given block returns a true value
- [ ] `#my_select` returns an enumerator if no block is given
- [ ] `#my_select` does not mutate the original array
- [ ] `#my_all?` returns true if the block never returns false or nil
- [ ] `#my_all?` returns false if the block returns false or nil
- [ ] `#my_all?` returns false if the block returns false or nil
- [ ] `#my_all?` does not mutate the original array
- [ ] `#my_all?` when no block or argument is given, it returns true when none of the collection members are false or nil
- [ ] `#my_all?` when no block or argument is given, it returns false when one of the collection members are false or nil
- [ ] `#my_all?` when a class is passed as an argument, it returns true if all of the collection is a member of such class
- [ ] `#my_all?` when a class is passed as an argument, it returns true if all of the collection is a member of such class
- [ ] `#my_all?` when a class is passed as an argument, it returns false if any of the collection is not a member of such class
- [ ] `#my_all?` when a Regex is passed as an argument, it returns true if all of the collection matches the Regex
- [ ] `#my_all?` when a Regex is passed as an argument, it returns false if any of the collection does not match the Regex
- [ ] `#my_all?` when a pattern other than Regex or a Class is given, it returns true if all of the collection matches the pattern
- [ ] `#my_all?` when a pattern other than Regex or a Class is given, it returns false if any of the collection does not match the pattern
- [ ] `#my_any?` returns false if the block ever returns a value false or nil
- [ ] `#my_any?` when no block or argument is given, it returns true if at least one of the collection is not false or nil
- [ ] `#my_any?` when no block or argument is given, it returns false if at least one of the collection is not true
- [ ] `#my_any?` when a class is passed as an argument, it returns true if at least one of the collection is a member of such class
- [ ] `#my_any?` when a class is passed as an argument, it returns true if at least one of the collection is a member of such class
- [ ] `#my_any?` when a Regex is passed as an argument, it returns true if any of the collection matches the Regex
- [ ] `#my_any?` when a Regex is passed as an argument, it returns false if none of the collection matches the Regex
- [ ] `#my_any?` when a pattern other than Regex or a Class is given, it returns false if none of the collection matches the pattern
- [ ] `#my_any?` when a pattern other than Regex or a Class is given, it returns true if any of the collection matches the pattern
- [ ] `#my_none?` returns true if the block never returns true for all elements
- [ ] `#my_none?` returns false if the block ever returns true for all elements
- [ ] `#my_none?` returns false if the block ever returns true for all elements
- [ ] `#my_none?` does not mutate the original array
- [ ] `#my_none?` when no block or argument is given, it returns true only if none of the collection members is true
- [ ] `#my_none?` when no block or argument is given, it returns false only if one of the collection members is true
- [ ] `#my_none?` when a class is passed as an argument, it returns true if none of the collection is a member of such class
- [ ] `#my_none?` when a class is passed as an argument, it returns true if none of the collection is a member of such class
- [ ] `#my_none?` when a class is passed as an argument, it returns false if any of the collection is a member of such class
- [ ] `#my_none?` when a Regex is passed as an argument, it returns true if none of the collection matches the Regex
- [ ] `#my_none?` when a Regex is passed as an argument, it returns false if any of the collection matches the Regex
- [ ] `#my_none?` when a pattern other than Regex or a Class is given, it returns true only if none of the collection matches the pattern
- [ ] `#my_none?` when a pattern other than Regex or a Class is given, it returns false only if one of the collection matches the pattern
- [ ] `#my_count` returns the number of items in enum through enumeration
- [ ] `#my_count` returns the number of items in enum through enumeration
- [ ] `#my_count` counts the number of items in enum that are equal to item if an argument is given
- [ ] `#my_map` returns a new array with the results of running a given block once for every element in enum
- [ ] `#my_map` returns an Enumerator if no block is given
- [ ] `#my_map` executes only the proc when both a block and a proc are given and returns a new array with the results of running the proc once for every element in enum.
- [ ] `#my_inject` when a block is given without an initial value, it combines all elements of enum by applying a binary operation, specified by a block
- [ ] `#my_inject` when a block is given with an initial value, it combines all elements of enum by applying a binary operation, specified by a block
- [ ] `#my_inject` when a symbol is specified without an initial value, it combines each element of the collection by applying the symbol as a named method
- [ ] `#my_inject` when a symbol is specified without an initial value, it combines each element of the collection by applying the symbol as a named method:range
- [ ] `#my_inject` when a symbol is specified with an initial value, it combines each element of the collection by applying the symbol as a named method
- [ ] `#my_inject` when a symbol is specified with an initial value, it combines each element of the collection by applying the symbol as a named method
- [ ] `#multiply_els` accepts an array as an argument and multiplies all the elements of the array together using `#my_inject`

Keep your heads up, and stay motivated. Happy coding 💻


### You can find me on slack 👉 [CY Kalu](https://microverse-students.slack.com/team/U01494U96FR) if you need further assistance or clarification. I'm always available and will be glad to help.

**Comment by mimipeshy** [comment]
#### STATUS: CHANGES REQUIRED :red_circle: :partying_face:
Hello team @pountzas  and @francisuloko    👋🏾 👋🏾 great job on this project

#### Highlights :white_check_mark: :white_check_mark:

- [ ]  You have implemented some of the feedback given by the previous TSE

However, before marking the project as complete please fix the following

- [x] Rubocop is failing, kindly check on this. You can use `rubocop --fix` to fix these errors
- [x] You have two enumerable files, kindly remove one and leave the final copy for review
- [x] In this review, I am giving feedback using enumerables2.rb, check out my comments below ⬇️ 

1. enumerables #my_each when a block is given when  self is a range it returns the range itself after calling the given block once for each element in self
2. enumerables #my_each when a block is given when self is a range it behaves like #each calls the given block once for each element in self
3. enumerables #my_each_with_index when a block is given when self is a range returns the range itself after calling the given block once for each element in self
4. enumerables #my_each_with_index when a block is given when self is a range behaves like #each_with_index calls the given block once for each element in self
5. enumerables #my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex
6. enumerables #my_all? when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern
7. enumerables #my_any? when a Regex is passed as an argument returns true if any of the collection matches the Regex
8. enumerables #my_any? when a pattern other than Regex or a Class is given returns true if any of the collection matches the pattern
9.  enumerables #my_none? when a Regex is passed as an argument returns false if any of the collection matches the Regex
10. enumerables #my_count returns the number of items in enum through enumeration::range
11. enumerables #my_inject raises a "LocalJumpError" when no block or argument is given

That's all :smile: kindly submit another review when done.

_Feel free to reach out to me on slack if you have any questions or something is unclear, or just comment on this PR and tag my name  (@mimipeshy) and I will be glad to help_
Regards
Peris Ndanu

Regards
Peris Ndanu

File: enumerables.rb [hunk 2]
```diff
@@ -0,0 +1,109 @@
+# Implementing  ruby enum methods
+module Enumerable
+  def my_each
```
Comment (smcommits) [inline enumerables.rb:4]: - [ ]  **#my_each when a block is given when self is a range returns the range itself after calling the given block once for each element in self. The following code should return the range itself**
```
range.my_each(&block)
```

File: enumerables.rb [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    return enum_for(:my_each) unless block_given?
+
+    length.times do |i|
+      yield to_a[i]
+    end
+    self
+  end
+
+  def my_each_with_index
```
Comment (smcommits) [inline enumerables.rb:13]: - [ ]  #my_each_with_index when a block is given when self is a range returns the range itself after calling the given block once for each element in self. The following code should return the range itself. 
```
range.my_each_with_index(&block)
```

File: enumerables.rb [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+  def my_select
+    return to_enum unless block_given?
+
+    temp = []
+    to_a.my_each { |item| temp.push(item) if yield(item) }
+    temp
+  end
+
+  def my_all?(arg = nil)
```
Comment (smcommits) [inline enumerables.rb:30]: - [ ]  **#my_all? when a Regex is passed as an argument returns true if all of the collection matches the Regex. The following code should return True. Instead, it is returning false.** 

```
expect(words.my_all?(/d/))
```

File: enumerables.rb [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+      to_a.my_each do |item|
+        return false unless item.is_a?(Numeric) || item.is_a?(Regexp)
+      end
+    else
+      to_a.my_each { |item| return false unless item }
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
```
Comment (smcommits) [inline enumerables.rb:45]: - [ ]  **#my_all? when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern. The following code should return true. Instead, it returns false**
```
words.my_any?(/d/))
```

File: enumerables.rb [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+      to_a.my_each do |item|
+        return true if item.is_a?(Integer) || item.is_a?(Regexp)
+      end
+    else
+      to_a.my_each { |item| return true if item }
+    end
+    false
+  end
+
+  def my_none?(arg = nil)
```
Comment (smcommits) [inline enumerables.rb:60]: - [ ]  **#my_none? when a Regex is passed as an argument returns false if any of the collection matches the Regex. The following should return false. Instead, it is returning true**

```
words.my_none?(/d/))
```

File: enumerables.rb [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+      to_a.my_each do |item|
+        return false if item.is_a?(Numeric) || item.is_a?(Regexp)
+      end
+    else
+      to_a.my_each { |item| return false if item }
+    end
+    true
+  end
+
+  def my_count(arg = nil)
```
Comment (smcommits) [inline enumerables.rb:75]: - [ ] **#my_count returns the number of items in enum through enumeration::range**

Input
```
range.my_count
```
Your output:
```
NameError:
        undefined local variable or method `length' for 5..50:Range
```

File: enumerables.rb [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+    temp = []
+    if my_proc
+      to_a.my_each { |item| temp << my_proc.call(item) }
+    else
+      to_a.my_each { |item| temp << yield(item) }
+    end
+    temp
+  end
+
+  def my_inject(sum = nil, symbol = nil)
```
Comment (smcommits) [inline enumerables.rb:100]: - [ ]  **#my_inject raises a "LocalJumpError" when no block or argument is given**


Input
```
array.my_inject
```
Your Output:
```
TypeError: nil is not a symbol nor a string
```

**Review by smcommits — CHANGES_REQUESTED** [review]
# Status: Changes Required
### Hello, team! You guys have done a great job. The methods are passing almost all of the test cases. But, you still haven't addressed the issues pointed out by the previous TSE. 

## Required Changes
#### Previous TSE pointed out some behavioral differences between the original enumerable methods and your enumerable methods. I can see that those behaviors are still not implemented. I have tried to provide a more comprehensive explanation using the inline comments above.

**Following variables were used for testing. Please refer to the inline comments above to understand how the variables are being used and what are the failing cases.**

```
ARRAY_SIZE = 100
LOWEST_VALUE = 0
HIGHEST_VALUE = 9

array = Array.new(ARRAY_SIZE) { rand(LOWEST_VALUE...HIGHEST_VALUE) } 
block = proc { |num| num < (LOWEST_VALUE + HIGHEST_VALUE) / 2 } 
words = %w[dog door rod blade] 
range = Range.new(5, 50) 
hash = { a: 1, b: 2, c: 3, d: 4, e: 5 } 
numbers = [1, 2i, 3.14] 
array_clone = array.clone 
```

# 
### Please don't hesitate to contact me on [Slack](https://microverse-students.slack.com/team/U017N7LLZ40) if you need assistance or believe I have overlooked something and made an unfair request.

**Review by mke2111 — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED 🔴 ♻️ 

## Hey Happy Coders, 👋🏽 
**This is a great project you have going on, well-done lads and thanks for implementing the changes requested earlier on**. However, there are a few issues to fix to get approved.

### REQUIRED CHANGES: ⚠️ 

- [ ] Check below on how to adjust the following methods. You can still use the sample variables given above.
```
my_all?
my_any?
my_none
my_inject

```
### `my_all?` Expected `true` but got `false`

- [ ] returns true if the block never returns false or nil. 
- [ ] when no block or argument is given returns true when none of the collection members are false or nil.
- [ ] when a class is passed as an argument returns true if all of the collection is a member of such class::Numeric
- [ ] when a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern


### `my_all?` Expected `false` but got `true`
- [ ] returns false if the block returns false or nil.
- [ ] when a Regex is passed as an argument returns false if any of the collection does not match the Regex
- [ ] returns false if the block returns false or nil::range.
- [ ] when no block or argument is given returns false when one of the collection members are false or nil.
- [ ] when a class is passed as an argument returns false if any of the collection is not a member of such class

### `my_any?` Expected `true` but got `false`

- [ ] returns true if the block ever returns a value other than false or nil
- [ ] returns false if the block ever returns a value false or nil::range
- [ ] when no block or argument is given returns true if at least one of the collection is not false or nil
- [ ] when a class is passed as an argument returns true if at least one of the collection is a member of such class::Numeric
- [ ] 

### `my_any?` Expected `false` but got `true`

- [ ] returns false if the block ever returns a value false or nil
- [ ] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
- [ ] when no block or argument is given returns false if at least one of the collection is not true
- [ ] when a class is passed as an argument returns true if at least one of the collection is a member of such class::Integer
- [ ] when a Regex is passed as an argument returns false if none of the collection matches the Regex

### `my_none?` Expected `true` but got `false`

- [ ] returns true if the block never returns true for all elements
- [ ] when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
- [ ] when no block or argument is given returns true only if none of the collection members is true
- [ ] when a Regex is passed as an argument returns true if none of the collection matches the Regex

### `my_none?` Expected `false` but got `true`

- [ ] returns false if the block ever returns true for all elements
- [ ] returns false if the block ever returns true for all elements::range
- [ ] when no block or argument is given returns false only if one of the collection members is true
- [ ] when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern

### `my_inject?` 
- [ ] This method is expected to raise a LocalJumpError, but instead got `TypeError: nil is not a symbol nor a string`

### Kindly make all the requested changes and submit another CRR. 💯 

**Cheers 🍹 and Happy coding 💻** 

## Reach me on slack for a chit-chat 🆘  Roy Mukuye

**Comment by francisuloko** [comment]
1. The first code reviewer asked us to use ruby doc test cases, and our methods passed all cases
2. The third code reviewer provided a different test case, and our methods passed everything except my_inject, of which ruby's inject also failed.
3. I really don't understand what is going on here. I feel everything contradicts previous code reviews. This is code review, play the role of a code reviewer and offer us a guide at least. Can you provide the test cases you used to try our methods please? @mke2111 @mimipeshy @cyonii @smcommits
Thank you.

**Comment by mke2111** [comment]
## Hi there, 👋🏽 

I used the same variables provided by @smcommits.
Your methods were (as of my review time) currently failing 28/79 tests.

I hope this will eliminate the contradiction you are currently perceiving.
For each test case, we have identified the pattern that leads to the given error. If you happen to find something not clear, please let us know. We are here to help each other.

**Comment by francisuloko** [comment]
That is great.

Did you notice any error in the failed cases he provided e.g words.my_any?(/d/)), words.my_none?(/d/)). What impact does that have on our code? Also, aren't our methods expected to be tried against ruby doc samples?

**Comment by mimipeshy** [comment]
### STATUS: CHANGES REQUIRED :writing_hand:
Hello team @francisuloko  and @pountzas -11, a great effort so far in this milestone. 👏🏾 

#### To Highlight

- [ ] Good job on following the correct GitHub flow ✔️  
- [ ] Your commit messages are well written in the present tense ✔️ 
- [ ] Your PR description is short and clear ✔️ 

#### To Improve
- [ ] As the previous TSE said, we all use the same variables to test the enumerables. Currently out of 76 test cases, 26 of those are still failing. 

#### `my_all?`
-  The main error here is in line 35 where you are passing a variable `arr` and it's not defined. Did you mean to pass `arg` instead? .Once you have worked on this, change the following to return the correct values

- [ ] returns false if the block returns false or nil::range Your method returns a true instead of false
- [ ]  when no block or argument is given returns false when one of the collection members are false or nil.  Your method returns a false instead of true
- [ ]  when a class is passed as an argument returns false if any of the collection is not a member of such class. Your method returns a false instead of true
- [ ] when a Regex is passed as an argument returns false if any of the collection does not match the Regex. Your method returns a false instead of true
- [ ] when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern. Your method returns a false instead of true

#### `my_any?`
-   The main error here is in line 48 where you are passing a variable `arr` and it's not defined. Did you mean to pass `arg` instead? .Once you have worked on this, change the following to return the correct values

- [ ] when a class is passed as an argument returns true if at least one of the collection is a member of such class::Integer. Your method returns a true instead of false
- [ ] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern. Your method returns a true instead of false

#### `my_none?`

- [ ] returns true if the block never returns true for all elements.  Your method returns a false instead of true
- [ ] returns false if the block ever returns true for all elements. Your method returns a true instead of false
- [ ] returns false if the block ever returns true for all elements::range. Your method returns a true instead of false
- [ ]  when no block or argument is given returns true only if none of the collection members is true.   Your method returns a false instead of true
- [ ] when no block or argument is given returns false only if one of the collection members is true. Your method returns a true instead of false
- [ ] when a Regex is passed as an argument returns true if none of the collection matches the Regex.  Your method returns a false instead of true
- [ ] when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern.   Your method returns a false instead of true
- [ ]  when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern. Your method returns a true instead of false


#### `my_inject?`

- [ ] raises a "LocalJumpError" when no block or argument is given
- [ ] when a symbol is specified with an initial value combines each element of the collection by applying the symbol as a named method
- [ ] when a symbol is specified with an initial value combines each element of the collection by applying the symbol as a named method::range



That's all :smile: kindly submit another review when done.

_Feel free to reach out to me on slack if you have any questions or something is unclear, or just comment on this PR and tag my name  (@mimipeshy) and I will be glad to help_
Regards
Peris Ndanu

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**Hello Nikos Pountzas & Francis Uloko** 
Good job with the improvements for this project :muscle:  :muscle: 
We still have to fix some issues before you can go ahead:

- [ ] You are not following to correct git-flow for this project, you must make development branch your main branch, then create a third branch with a descriptive name (it can be "feature"), then make all your project there and finally make a pull request from your feature branch to the development branch.
- [ ] Fix rubocop offenses
- [ ] #my_all? returns false if the block returns false or nil::range
    `(1..3).all?(&proc{|x| x%2==0}) == (1..3).my_all?(&proc{|x| x%2==0})`
- [ ] #my_all? when no block or argument is given returns false when one of the collection members are false or nil
      `[true,[true],false].all? == [true,[true],false].my_all?`
- [ ] #my_all? when a class is passed as an argument returns false if any of the collection is not a member of such class       
      `['word',1,2,3].all?(Integer) == ['word',1,2,3].my_all?(Integer)`
- [ ] #my_all? when a Regex is passed as an argument returns false if any of the collection does not match the Regex
      `['car', 'cat'].all?(/t/) == ['car', 'cat'].my_all?(/t/)`
- [ ] #my_all? when a pattern other than Regex or a Class is given returns false if any of the collection does not match the pattern
      `[5,5,[5]].all?(5) == [5,5,[5]].my_all?(5)`
- [ ] #my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class::Numeric
      `[1.1,'',[]].any?(Numeric) == [1.1,'',[]].my_any?(Numeric)`
- [ ] #my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class::Integer
      `[1,'',[]].any?(Numeric) == [1,'',[]].my_any?(Integer)`
- [ ] #my_none? when no block or argument is given returns false only if one of the collection members is true
      `[false, nil,[]].none? == [false, nil,[]].my_none?`
- [ ] #my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
      `[true,[]].none?(String) == [true,[]].my_none?(String)`
- [ ] #my_none? when a class is passed as an argument returns true if none of the collection is a member of such class::Numeric 
      `[true,[]].none?(Numeric) == [true,[]].my_none?(Numeric)`
- [ ] #my_none? when a class is passed as an argument returns false if any of the collection is a member of such class
      `['',[]].none?(String) == ['',[]].my_none?(String)`
- [ ] #my_none? when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern
      `[5,'dog','car'].none?(5) == [5,'dog','car'].my_none?(5)`
- [ ] #my_inject when a block is given with an initail value combines all elements
of enum by applying a binary operation, specified by a block
      `(1..3).inject(4) { |prod, n| prod * n } == (1..3).my_inject(4) { |prod, n| prod * n }`
- [ ] #my_inject when a symbol is specified without an initail value combines each
element of the collection by applying the symbol as a named method
      `[1,2,3].inject(:+) == [1,2,3].my_inject(:+)`
- [ ] #my_inject when a symbol is specified without an initail value combines each
element of the collection by applying the symbol as a named method:range
      `(1..9).inject(:+) == (1..9).my_inject(:+)`
- [ ] #multiply_els accepts an array as an argument and multiplies all the elements of the array together using #my_inject      
    `multiply_els([1,2,3]) == 6`
 
Please request another review once you are ready! :+1: 
 
Happy coding :relaxed:
 
Oscar Andrés Russi Porras

**Review by MkrtichSargsyan — APPROVED** [review]
**Status:** Approved  :hamster: :rocket:

Congratulations on your hard work, your project is now approved, thank you for making the changes that previous reviewers suggested.

## Optional

there are still some rubocop errors that showing in my local machine, but I'm approving it because it's the 7th review.

make sure you correct them before merging.

> ps: . If you have any questions about my review or need help, you can message me on slack ( @Mkrtich Sargsyan )

**Pull request merged**